app/services/invite_schools.rb

```diff
@@ -48,6 +76,21 @@ def email_expiry_date
     NominationEmail::NOMINATION_EXPIRY_TIME.from_now.strftime("%d/%m/%Y")
   end
 
+  # Notify gives us a 3000/min rate limit. Limit this to 1800/min to allow for other calls and leeway
+  def rate_limit
```

Comment:
We use `deliver_now`, which I think is synchronous. Do we ever actually get close to 30 requests per second this way?

Reply:
And secondly, do we need to test it? Maybe less sophisticated `sleep(0.1.second)` would do the trick too and be much simpler to understand?

Reply:
Not sure what the realised throughput is on Notify, I suspect the answer is "it depends". I went for this over a simple `sleep` since it's going to take _aaages_ already and I wanted to keep it as quick as possible. Happy to remove if you think this is particularly unclear though.

Reply:
How long does it take per email? I would think for each call, we make an api call to notify, we wait for it to finish, and then we move on to the next one.

And I find it really unlikely that we could do 50 api calls to notify synchronously in a single second. Which is probably why it is taking ages already.

Reply:
I think if we were ever gonna hit the rate limit on notify we would have when we sent school invite emails - or did we do it in batches smaller than 3k at all times?

Reply:
@kicferk1 how does that look now?

Reply:
Much better, thanks :)